SharpRepository.Repository/RepositoryBase.cs

```diff
@@ -1278,12 +1303,14 @@ public TResult Max<TResult>(Expression<Func<T, bool>> predicate, Expression<Func
 
         private bool RunAspect(Func<RepositoryActionBaseAttribute, bool> action)
         {
-            return _aspects.All(action);
+            return _aspects.Values
+                .Where(a => a.Enabled)
+                .All(action);
         }
 
         private void RunAspect(Action<RepositoryActionBaseAttribute> action)
         {
-            foreach (var attribute in _aspects)
+            foreach (var attribute in _aspects.Values.Where(a => a.Enabled))
```

Comment:
Oh, I've missed that point about ordering. I will do changes on Monday and check if I can write some unit tests